src/components/plots/gridPlot.ts

```diff
@@ -48,23 +53,46 @@ export module Plot {
      */
     public project(attrToSet: string, accessor: any, scale?: Scale.AbstractScale<any, any>) {
       super.project(attrToSet, accessor, scale);
+
+      if (attrToSet === "x") {
+        super.project("x1", (d: any, i: number, u: any, m: Plot.PlotMetadata) => {
+          return scale.scale(this._projections["x"].accessor(d, i, u, m));
+        });
+      }
```

Comment:
Needs to be the other way around, because AbstractXY plot leverages x/y directly